File: GamingPlatform/wwwroot/js/morpion.js
// ===============================
// CONNEXION SIGNALR
// ===============================
const connection = new signalR.HubConnectionBuilder()
    .withUrl("/morpionHub")
    .build();

// ===============================
// ELEMENTS DOM
// ===============================
const cells = document.querySelectorAll(".cell");
const status = document.getElementById("status");
const restartBtn = document.getElementById("restartBtn");

// ===============================
// RECEPTION ETAT DU JEU
// ===============================
connection.on("UpdateGame", game => {

    // Mise à jour du plateau
    game.board.forEach((value, index) => {
        cells[index].textContent = value;
        cells[index].classList.toggle("disabled", value !== "");
    });

    // Etat de la partie
    if (game.finished) {
        status.textContent =
            game.winner === "Draw"
                ? "Match nul 🤝"
                : `Victoire de ${game.winner} 🎉`;

        restartBtn.style.display = "inline-block";
    } else {
        status.textContent = `Tour du joueur ${game.currentPlayer}`;
        restartBtn.style.display = "none";
    }
});

// ===============================
// CLICK SUR UNE CASE
// ===============================
cells.forEach(cell => {
    cell.addEventListener("click", () => {
        const index = parseInt(cell.dataset.index);

        connection.invoke("PlayMove", index)
            .catch(err => console.error("Erreur SignalR :", err));
    });
});

// ===============================
// REJOUER
// ===============================
restartBtn.addEventListener("click", () => {
    connection.invoke("ResetGame")
        .catch(err => console.error("Erreur reset :", err));
});

// ===============================
// CONNEXION AU SERVEUR
// ===============================
connection.start()
    .then(() => {
        status.textContent = "En attente d’un autre joueur...";
    })
    .catch(err => {
        console.error(err);
        status.textContent = "Erreur de connexion au serveur";
    });

File: Platform/Models/Morpion.cs
using System.Linq;

namespace GamingPlatform.Models
{
    public class Morpion
    {
        public string[] Board { get; private set; } = new string[9];
        public string CurrentPlayer { get; private set; } = "X";
        public bool Finished { get; private set; }
        public string? Winner { get; private set; }

        public Morpion()
        {
            Reset();
        }

        public void Play(int index)
        {
            // Sécurités STRICTES
            if (Finished)
                return;

            if (index < 0 || index >= 9)
                return;

            if (!string.IsNullOrEmpty(Board[index]))
                return;

            Board[index] = CurrentPlayer;

            CheckWinner();

            if (!Finished)
                CurrentPlayer = CurrentPlayer == "X" ? "O" : "X";
        }

        public void Reset()
        {
            for (int i = 0; i < 9; i++)
                Board[i] = "";

            CurrentPlayer = "X";
            Finished = false;
            Winner = null;
        }

        private void CheckWinner()
        {
            int[,] wins =
            {
                {0,1,2},{3,4,5},{6,7,8},
                {0,3,6},{1,4,7},{2,5,8},
                {0,4,8},{2,4,6}
            };

            for (int i = 0; i < wins.GetLength(0); i++)
            {
                string a = Board[wins[i, 0]];
                string b = Board[wins[i, 1]];
                string c = Board[wins[i, 2]];

                if (!string.IsNullOrEmpty(a) && a == b && b == c)
                {
                    Winner = a;
                    Finished = true;
                    return;
                }
            }

            // Match nul
            if (Board.All(x => !string.IsNullOrEmpty(x)))
            {
                Winner = "Draw";
                Finished = true;
            }
        }
    }
}
